packages/react-aria-components/stories/GridList.stories.tsx

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2022 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import {Button, GridList, GridListItem, GridListItemProps} from 'react-aria-components';
+import {classNames} from '@react-spectrum/utils';
+import React from 'react';
+import styles from '../example/index.css';
+
+export default {
+  title: 'React Aria Components'
+};
+
+export const GridListExample = (args) => (
+  <GridList
+    {...args}
+    className={styles.menu} 
+    aria-label="test gridlist"
+    style={{
+      width: 300,
+      height: 300,
+      display: 'grid',
+      gridTemplate: args.layout === 'grid' ? 'repeat(3, 1fr) / repeat(3, 1fr)' : 'auto / 1fr',
+      gridAutoFlow: 'row'
+    }}>
+    <MyGridListItem>1,1 <Button>Actions</Button></MyGridListItem>
```

Comment:
do items know what is their index in the data source?
do items know what is their position in the visible viewport/row?